lib/spack/spack/test/conftest.py

```diff
@@ -690,3 +690,20 @@ def repo_for_pkg(self, name):
         import collections
         Repo = collections.namedtuple('Repo', ['namespace'])
         return Repo('mockrepo')
+
+##########
+# Specs of various kind
```

Comment:
I think the following would be clearer:

> Specs which violate constraints specified with the "conflicts" directive in the "conflict" package

Other than that this looks good to me.

Reply:
@alalazo you've been pretty busy - any chance this fell off your radar?

Reply:
@scheibelp I updated the docstring of the fixture, but I am not sure you meant that or the comment:
```python
# Specs of various kind
```
I left the comment untouched because it is meant to divide this particular fixture from the previous:
```python
# Mock packages
```
and, if we maintain the adjective generic enough, we can add at this point in the file fixtures that return specs with different properties. Let me know if your intention was to change the comment, or remove it altogether.

Reply:
Actually that makes sense as you described it. This comment can be more generic since there is a docstring for `conflict_spec` (thanks for updating the docstring!)